### PR TITLE
Bug fix to many request when round info is indexed

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: process/block/metrics.go [hunk 1]
```diff
@@ -211,15 +211,17 @@ func indexRoundInfo(
...
+		go indexerHandler.SaveRoundsInfos([]indexer.RoundInfo{roundInfo})
 		return
 	}
 
 	lastBlockRound := lastHeader.GetRound()
 	currentBlockRound := header.GetRound()
 	roundDuration := calculateRoundDuration(lastHeader.GetTimeStamp(), header.GetTimeStamp(), lastBlockRound, currentBlockRound)
+
+	roundsInfos := make([]indexer.RoundInfo, 0)
+	roundsInfos = append(roundsInfos, roundInfo)
```
Comment (SebastianMarian) [inline process/block/metrics.go:224]: With this implementation the highest round would be first in this slice, and then the others would be sorted from lower to higher. It is ok ?
I meant that if let say currentBlockRound = 10 and lastBlockRound = 6 the results of the roundsInfos slice will be:
10, 7, 8, 9

File: core/indexer/elasticsearchDatabase.go [hunk 2]
```diff
@@ -289,35 +288,44 @@ func (esd *elasticSearchDatabase) getMiniblocks(header data.HeaderHandler, body
...
-	if err != nil {
-		log.Debug("indexer: marshal", "error", "could not marshal signers indexes")
-		return
+	for _, info := range infos {
+		serializedRoundInfo, meta := serializeRoundInfo(info)
+
+		buff.Grow(len(meta) + len(serializedRoundInfo))
+		_, err := buff.Write(meta)
+		if err != nil {
+			log.Warn("indexer: serialized round info", "error", err.Error())
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply core/indexer/elasticsearchDatabase.go:301]: +1

File: core/indexer/elasticsearchDatabase.go [hunk 3]
```diff
@@ -289,35 +288,44 @@ func (esd *elasticSearchDatabase) getMiniblocks(header data.HeaderHandler, body
...
+		}
+		_, err = buff.Write(serializedRoundInfo)
+		if err != nil {
+			log.Warn("indexer: serialized round info meta", "error", err.Error())
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply core/indexer/elasticsearchDatabase.go:305]: +1

File: core/indexer/elasticsearchDatabase.go [hunk 4]
```diff
@@ -289,35 +288,44 @@ func (esd *elasticSearchDatabase) getMiniblocks(header data.HeaderHandler, body
...
-		return
+	for _, info := range infos {
+		serializedRoundInfo, meta := serializeRoundInfo(info)
+
+		buff.Grow(len(meta) + len(serializedRoundInfo))
+		_, err := buff.Write(meta)
+		if err != nil {
+			log.Warn("indexer: serialized round info", "error", err.Error())
+		}
+		_, err = buff.Write(serializedRoundInfo)
```
Comment (sasurobert) [inline core/indexer/elasticsearchDatabase.go:303]: empty line before

File: core/indexer/elasticsearchDatabase.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:301]: done.

File: core/indexer/elasticsearchDatabase.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:305]: done.

File: process/block/metrics.go (hunk 1 above)
Comment (miiu96) [reply process/block/metrics.go:224]: that's how it is but is not important order of elements in slice in this case /

File: core/indexer/elasticsearchDatabase.go (hunk 4 above)
Comment (miiu96) [reply core/indexer/elasticsearchDatabase.go:303]: done.

**Pull request merged**